### PR TITLE
Integration Tests only: add static route, snmp_user, snapshot and hsrp it cases

### PR DESCRIPTION
**Pull request by saichint**
##### SUMMARY

IT cases for hsrp, snapshot, snmp_user and static_route

##### ISSUE TYPE
 - IT Pull Request

##### COMPONENT NAME
- nxos_static_route
- nxos_hsrp
- nxos_snapshot
- nxos_snmp_user

##### ANSIBLE VERSION

```
ansible 2.4.0 (devel 54cd641df4) last updated 2017/07/19 18:06:44 (GMT -400)
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /root/p2/Projects/nxos_ansible/p1_ansible/lib/ansible
  executable location = /root/p2/Projects/nxos_ansible/p1_ansible/bin/ansible
  python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]
```

##### ADDITIONAL INFORMATION

This PR is for adding IT cases for nxos_snapshot, nxos_snmp_user, nxos_static_route and nxos_hsrp components. There are no code changes.

```

```

### CODE REVIEW

File: test/integration/targets/nxos_snapshot/tests/common/sanity.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    snapshot_name: test_snapshot
+    description: Ansible
+    provider: "{{ connection }}"
+
+- name: delete snapshot
+  nxos_snapshot:
+    action: delete_all
+    provider: "{{ connection }}"
+
+- debug: msg="END TRANSPORT:{{ connection.transport }} nxos_rollback sanity test"
```
Comment (Qalthos) [inline test/integration/targets/nxos_snapshot/tests/common/sanity.yaml]: nxos_snapshot

**Review by Qalthos — CHANGES_REQUESTED** [review]

File: test/integration/targets/nxos_snapshot/tests/common/sanity.yaml (hunk 1 above)
Comment (saichint) [reply test/integration/targets/nxos_snapshot/tests/common/sanity.yaml]: what is the requested change here? All it says is "nxos_snapshot" which does not give me anything to go on.

Comment (saichint) [reply test/integration/targets/nxos_snapshot/tests/common/sanity.yaml]: found the typo. sorry did not catch it first time

**Comment by Qalthos** [comment]
It looks like `nxos_snapshot` isn't working on titanium, here's the response:

`"ERROR: Failed to change group ownership on snapshot\nExecuting 'show interface'... Done\nExecuting 'show ip route summary vrf all'... Done\nExecuting 'show ipv6 route summary vrf all'... Done\nFeature 'bgp' not enabled, skipping...\nFeature 'eigrp' not enabled, skipping...\nFeature 'eigrp' not enabled, skipping...\nExecuting 'show vpc'... Done\nFeature 'ospf' not enabled, skipping...\nFeature 'ospfv3' not enabled, skipping...\nFeature 'isis' not enabled, skipping...\nFeature 'rip' not enabled, skipping...\nSyntax error while parsing 'source sys/snapshot_ext.py section-cmds 1'\n\n\nCmd exec error.\nFailed to execute source sys/snapshot_ext.py section-cmds 1\nSnapshot 'test_snapshot1' created\n"`

To be fair, cli responds the exact same way, but nxapi is a lot clearer about what it decides is an error or not

**Comment by saichint** [comment]
@Qalthos Added skip the tests for titanium now.

File: test/integration/targets/nxos_snapshot/tests/common/sanity.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      comparison_results_file: compare_snapshots.txt
+      compare_option: summary
+      path: '.'
+      provider: "{{ connection }}"
+
+  - name: delete snapshot
+    nxos_snapshot:
+      action: delete_all
+      provider: "{{ connection }}"
+  when: (titanium is defined) and not ((titanium | search('true')))
```
Comment (Qalthos) [inline test/integration/targets/nxos_snapshot/tests/common/sanity.yaml]: This doesn't have to be skipped on both transports, cli is fine. a check in nxapi/sanity.yaml would be more precise.

**Review by Qalthos — CHANGES_REQUESTED** [review]

File: test/integration/targets/nxos_snapshot/tests/common/sanity.yaml (hunk 2 above)
Comment (saichint) [reply test/integration/targets/nxos_snapshot/tests/common/sanity.yaml]: just hold the PR, until I retest this and clean it for nxapi only. Having difficulty in getting the proper when condition.

**Comment by saichint** [comment]
@Qalthos Added proper titanium condition and the nxapi exclusion. The previous titanium fact was not correct and that was causing issues, it is cleaned in the prepare_nxos_tests as well. bgp tests which used this are also fixed. Done with all my changes for this PR.

**Review by Qalthos — APPROVED** [review]

**Comment by abadger** [comment]
Cherry-picked to stable-2.4 for 2.4.2beta2

**Pull request merged**